### PR TITLE
Add wishlist cart and refactor

### PR DESCRIPTION
**Pull request by varun13169**
**All files are relevant ones.**

Profducts Page
image

Cart Page
image

Wishlist Page
image

### CODE REVIEW

File: src/pages/cartPage/cartPageUtils.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
+import axios from "axios";
+const incrementItemInCartAndSetCart = ({ itemDetails, cart, setCart }) => {
+  // setCart((cart) =>
+  //   cart.map((e) => {
+  //     e["id"] === itemDetails["id"] ? { ...e, qty: e.qty + 1 } : e;
+  //   })
+  // );
```
Comment (archit7448) [inline src/pages/cartPage/cartPageUtils.js:7]: remove commented codes

**Review by archit7448 — APPROVED** [review]
logic implementation is nice  ,  proper structure 
  looks good to merge

File: src/pages/cartPage/cartPageUtils.js (hunk 1 above)
Comment (varun13169) [reply src/pages/cartPage/cartPageUtils.js:7]: This is a feature, it's not working as of now will be asking the same in DSH.

File: src/pages/cartPage/cartPageUtils.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  (async () => {
+    try {
+      let res = await axios.post(
+        "/api/user/cart/" + itemDetails._id,
+        payload,
+        config
+      );
+      setCart((cart) => res.data.cart);
+    } catch (err) {
+      console.log(err);
```
Comment (purvasharmadev) [inline src/pages/cartPage/cartPageUtils.js:29]: Remove console.log.

**Review by purvasharmadev — APPROVED** [review]
code is clean and understandable, good job with logic too, good to go.

File: src/pages/cartPage/CartPage.jsx [hunk 3]
```diff
@@ -1,71 +1,102 @@
...
+                    {Number(getTotalSalePriceOfCart(cart)) +
+                      Number(getDeliveryCharges(cart))}
+                  </p>
+                </div>
+                <hr />
+                <p>
+                  You will save ${getTotalDiscountOnCart(cart)} on this order
+                </p>
+
+                <a
```
Comment (santhoshpemmaka) [inline src/pages/cartPage/CartPage.jsx:85]: If possible convert a tag into a Link tag.

**Comment by santhoshpemmaka** [comment]
UI looking Great and code structure clean 💯 . Good to merge 👍

**Review by santhoshpemmaka — APPROVED** [review]

File: src/pages/cartPage/cartPageUtils.js (hunk 2 above)
Comment (varun13169) [reply src/pages/cartPage/cartPageUtils.js:29]: Will be replacing with logger

File: src/pages/cartPage/CartPage.jsx (hunk 3 above)
Comment (varun13169) [reply src/pages/cartPage/CartPage.jsx:85]: will make it part of refactoring.

**Pull request merged**